core/capabilities/ccip/oraclecreator/plugin.go

```diff
@@ -284,33 +290,57 @@ func (i *pluginOracleCreator) createReadersAndWriters(
 
 	contractReaders := make(map[cciptypes.ChainSelector]types.ContractReader)
 	chainWriters := make(map[cciptypes.ChainSelector]types.ChainWriter)
-	for _, chain := range i.chains.Slice() {
-		chainSelector, err1 := i.getChainSelector(chain.ID().Uint64())
+	for relayID, relayer := range i.relayers {
+		chainID, err1 := decimal.NewFromString(relayID.ChainID)
 		if err1 != nil {
-			return nil, nil, err1
+			return nil, nil, fmt.Errorf("error parsing chain ID, expected big int: %s %w", relayID.ChainID, err1)
+		}
+
+		chainSelector, err1 := i.getChainSelector(chainID.BigInt().Uint64())
+		if err1 != nil {
+			return nil, nil, fmt.Errorf("failed to get chain selector from chain ID %s: %w", chainID.String(), err1)
+		}
+
+		chainReaderConfig, err1 := getChainReaderConfig(chainID.BigInt().Uint64(), destChainID, homeChainID, ofc, chainSelector)
+		if err1 != nil {
+			return nil, nil, fmt.Errorf("failed to get chain reader config: %w", err1)
 		}
 
-		chainReaderConfig := getChainReaderConfig(chain.ID().Uint64(), destChainID, homeChainID, ofc, chainSelector)
-		cr, err1 := createChainReader(i.lggr, chain, chainReaderConfig, pluginType)
+		// TODO: context.
+		cr, err1 := relayer.NewContractReader(context.Background(), chainReaderConfig)
 		if err1 != nil {
 			return nil, nil, err1
 		}
 
-		if err2 := bindContracts(chain, cr, config, destChainID); err2 != nil {
-			return nil, nil, err2
+		if chainID.BigInt().Uint64() == destChainID {
```

Comment:
ChainID has four different types, `decimal` `bigInt` and `uint64` `string` within just a few lines of code.
We need some CLL standard and fix all this :S

Reply:
I can probably switch from `decimal` to `big.Int.SetString` instead, will do that to reduce overhead.